### PR TITLE
feat(uirefresh): component refresh - trade box [2/2]

### PR DESCRIPTION
**Pull request by moo-onthelawn**
[Figma](https://www.figma.com/design/fMQodZKzn9B5aZTN5G0fLJ/dYdX-%E2%80%BA-Web?node-id=41621-3229&node-type=frame&t=XbiUqbZ76UFDIHwr-0) 
1. [Flagged] Updates margin selector
2. [Flagged] Moves target leverage selector to be a leverage slider very similar to leverage in cross margin orders
3. Fixes invalid dom nesting issue in HorizontalPanel's CollapsibleTabs (due to rendering a Button in `slotTrigger` of `TradeTableSettings`)

To follow up on:
- Add a tooltip to "Margin Mode" - still waiting for copy

| | Flag Off | Flag On |
| ---- | ---- | ---- |
| Web - toggleable | Screenshot 2024-10-24 at 1 09 03 PM | Screenshot 2024-10-24 at 1 04 36 PM |
| Web - untoggleable | Screenshot 2024-10-24 at 1 08 38 PM | Screenshot 2024-10-24 at 1 04 06 PM |
| Mobile - toggleable | Screenshot 2024-10-24 at 1 09 15 PM | Screenshot 2024-10-24 at 12 56 01 PM |
| Mobile - untoggleable | Screenshot 2024-10-24 at 1 09 23 PM | Screenshot 2024-10-24 at 12 56 13 PM |

---

## Views

- `<TargetLeverageButton>`
  - Small cleanup to use more specific target leverage selector

- `<TradeSizeInputs>`
  - [Flagged] Conditionally render `TargetLeverageInput` component when necessary

- New: `<TargetLeverageInput>`
  - Basically copied over the code from `AdjustTargetLeverageForm`. Did not bother refactoring to share this component since that component can be deprecated when `uiRefresh` is launched
  - Slightly updated to add debouncing logic

- `<MarginAndLeverageButtons>`
  - [Flagged] Update to not render TargetLeverageButton when flag on

- `<MarginModeSelector>`
  - [Flagged] Updated rendering when uiRefresh enabled

- `<TradeBoxOrderView>`
  - [Flagged] styling

- `<TradeTableSettings>`
  - Reorder Isolated / Cross to match with toggle order in trade box
  - Fix DOM nesting error

## Components

- `<ToggleGroup>`
  - Add some styling for disabled cursors

## Functions/Clients

- `lib/abacus/index.ts`
  - Clear targetLeverage when necessary since it's now a user input on the trade form

---

### CODE REVIEW

File: src/components/ToggleGroup.tsx [hunk 1]
```diff
@@ -69,7 +71,7 @@ export const ToggleGroup = forwardRefFn(
         <$ToggleButton
           size={size ?? (isTablet ? ButtonSize.Small : ButtonSize.XSmall)}
           shape={shape}
-          disabled={item.disabled}
+          disabled={item.disabled ?? disabled}
```
Comment (tyleroooo) [inline src/components/ToggleGroup.tsx:74]: shouldn't this be || ?

File: src/lib/abacus/index.ts [hunk 2]
```diff
@@ -199,6 +200,10 @@ class AbacusStateManager {
       this.setTradeValue({ value: null, field: TradeInputField.leverage });
     }
 
+    if (uiRefresh && needsTargetLeverage) {
```
Comment (tyleroooo) [inline src/lib/abacus/index.ts:203]: Why not just clear them all the time? I worry about edge cases where target leverage is entered but then needsTargetLeverage becomes false before this clear function is called and we just leave it there forever.

Comment (moo-onthelawn) [reply src/lib/abacus/index.ts:203]: ah good point- will update!

**Review by tyleroooo — APPROVED** [review]
